1.regression_modelling/scripts/0.split_data.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from pycytominer import feature_select
+
+
+# ## Load in normalized profiles and concat into one dataframe
+
+# In[2]:
+
+
+# path to big drive where data is located (large)
+drive_path = pathlib.Path("../../../../media/18tbdrive/")
+
+# set path to folder in image profiling repo with the normalized profiles
+norm_path = pathlib.Path(
+    drive_path / "Github_Repositories/nuclear_speckle_image_profiling/4.preprocess_features/data/single_cell_profiles"
+).resolve(strict=True)
+
+# load all normalized parquet files
+files = norm_path.glob("*_sc_normalized.parquet")
+dfs = [pd.read_parquet(file) for file in files]
+
+# concatenate them into one data frame
+combined_df = pd.concat(dfs, ignore_index=True)
+
+# perform feature selection to drop any columns that have NaN (avoid downstream issues)
+combined_df = feature_select(
+    combined_df,
+    operation="drop_na_columns",
+    na_cutoff=0
+)
+
```

Comment:
Feature selection maybe belongs in another notebook used during processing?

Reply:
Since we are using normalized data to generate these models, there is a likelihood that there are NaNs in the data.

Instead of using a different method, I use pycytominer to clear out columns with NaNs prior to training. So it would probably best fit in this notebook since it is a small step in the process.

Reply:
Question: if removing columns (features) that have NaNs, is this then feature selection? The model is generated on feature selected data and not the normalized data?

Reply:
It is generated on the normalized data because we don't perform any variance thresholding or correlation thresholding like we traditionally would on a feature-selected dataset.

We are just removing columns with Nans, not trying to find the least redundant features.

Reply:
Ahhh, gotcha gotcha!